### ISSUE TITLE
Add a mechanism to trace edgedb queries and their performance

### ISSUE THREAD

**Issue by andrewpatto** [umccr/elsa-data#377]
In practice we are going to start a phase where we start to see "real" database queries against more realistic data - and we are going to start to ask "why is that slow?".

Can we put some sort of intereceptor on the edgedb client that does wall clock measurements and logs TRACE messages?

Or is there something we can get out of the EdgeDb deployment itself that we could easily correlate with UI activity (i.e at time XYZ - go and look at the edgedb logs).

Just need some plausible mechanism here to help us isolate are we missing indexes, doing double queries etc.

**Comment by DoxasticFox** [comment]
Another approach could be to use the PostgreSQL's `pg_stat_statements` module, seeing as EdgeDB is backed by PostgreSQL.

I'm guessing you use Amazon RDS. [It sounds like it's enabled by default there](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.UsingDashboard.AnalyzeDBLoad.AdditionalMetrics.PostgreSQL.html). If so, query performance is already being tracked, and you can retrieve the statistics by issuing SQL queries.

**Comment by andrewpatto** [comment]
Yep - Marco has added some AWS stats monitoring on the RDS. I think the difficulty will be that presumably EdgeDb turns its queries into multiple actual SQL - so might be difficult to split them out as to what was caused by what. I mean getting some sort of correlation id that extends all the way from our EdgeQl execution - through into the backend SQL monitoring would be brilliant..

**Comment by andrewpatto** [comment]
For this ticket I was thinking something super simple - like literally some sort of code pattern that logs - whether by interceptor or just even a code convention we use.

-- starting EdgeQl 'extract_patient_ids'
..
-- 'extract_patient_ids' took 151ms

That way when users say something is slow we could look at the logs and find edgeql where time > 1second (say).

I mean we should probably log things with a proper JSON format and then we can literally do the looking with Cloudwatch query.
